Runtime/MeshChainRenderer.cs

```diff
@@ -0,0 +1,459 @@
+                m_Color.alphaKeys = alphaKeys;
+                UpdateColors();
+            }
+        }
+
+        /// <summary>
+        /// Set the color at the end of the line.
+        /// </summary>
+        public Color colorEnd
+        {
+            get { return m_Color.Evaluate(1); }
+            set
+            {
+                var colorKeys = m_Color.colorKeys;
+                var alphaKeys = m_Color.alphaKeys;
+                var lastColorIndex = colorKeys.Length - 1;
+                var lastAlphaIndex = alphaKeys.Length - 1;
+                var flatColor = value;
+                flatColor.a = 1.0f;
+                colorKeys[lastColorIndex].color = flatColor;
+                alphaKeys[lastAlphaIndex].alpha = value.a;
+                m_Color.colorKeys = colorKeys;
+                m_Color.alphaKeys = alphaKeys;
+                UpdateColors();
+            }
+        }
+
+        /// <summary>
+        /// Resets the width to a straight curve at the given value
+        /// </summary>
+        /// <param name="newWidth">The new width for the curve to display</param>
+        public void SetTotalWidth(float newWidth)
+        {
+            m_Width = newWidth;
+            m_WidthCurve = new AnimationCurve(new Keyframe(0, 1.0f));
+            UpdateWidth();
+        }
+
+        /// <summary>
+        /// Resets the color to a single value
+        /// </summary>
+        /// <param name="newColor">The new color for the curve to display</param>
+        public void SetTotalColor(Color newColor)
+        {
+            var flatColor = newColor;
+            flatColor.a = 1.0f;
+            m_Color = new Gradient
+            {
+                alphaKeys = new[] { new GradientAlphaKey(newColor.a, 0), new GradientAlphaKey(newColor.a, 1), },
+                colorKeys = new[] { new GradientColorKey(flatColor, 0), new GradientColorKey(flatColor, 1) },
+                mode = GradientMode.Blend
+            };
+            UpdateColors();
+        }
+
+        void OnValidate()
+        {
+            SetupMeshBackend();
+            if (NeedsReinitialize())
+            {
+#if UNITY_EDITOR
+                UnityEditor.EditorApplication.delayCall += EditorCheckForUpdate;
+#endif
+            }
+            else
+            {
+                EditorCheckForUpdate();
+            }
+        }
+
+        /// <summary>
+        /// Cleans up the visible interface of the MeshRenderer by hiding unneeded components
+        /// Also makes sure the animation curves are set up properly to defaults
+        /// </summary>
+        void SetupMeshBackend()
+        {
+            if (m_MeshRenderer == null)
+                m_MeshRenderer = GetComponent<MeshRenderer>();
+
+            if (m_MeshRenderer.hideFlags == HideFlags.None)
+                m_MeshRenderer.hideFlags = HideFlags.HideInInspector;
+
+            var meshFilter = GetComponent<MeshFilter>();
+            if (meshFilter.hideFlags == HideFlags.None)
```

Comment:
This was a nasty bug. Setting `.hideFlags` dirties the object. I was first checking if the hideflags were anything but `HideInInspector`, but for some reason, on import, the hideflags on these components come back as `HideInHierarchy` when the file clearly shows a value of `2`, which is `HideInInspector`. This seems to work, but it will fail if the hideflags are something other than None.